src/yelp-application.c

```diff
@@ -246,9 +246,13 @@ yelp_application_cmdline (GApplication     *app,
             cur = g_get_current_dir ();
             base = g_file_new_for_path (cur);
             new = g_file_resolve_relative_path (base, (*arguments)[i]);
-            newuri = g_file_get_uri (new);
-            g_free ((*arguments)[i]);
-            (*arguments)[i] = newuri;
+
+            /* Hack to handle --gapplication-service flag. */
```

Comment:
Yeah, it's awful, but this piece of code wants to change all the parameters (that are likely relative paths) to file:// URIs before they are sent to the remote application, so we can't move the chained-up call. The comment could be better though - something like `Hack to avoid mangling --gapplication-service flag`.

Probably the even better solution would be to add another flag to yelp about the base directory for the relative paths and add this flag here. So you call `yelp thispage.mallard dir/thatpage.mallard` and it will send `--base-dir=$PWD thispage.mallard dir/thatpage.mallard` to the remote application, so the remote application can convert the paths to URIs.

But the current hack "works".